### ISSUE TITLE
Pyright's `include` setting is apparently used by pylances refactoring functionality

### ISSUE THREAD

**Issue by lorenzwalthert** [microsoft/pylance-release#5605]
## Environment data

-   Pylance version: v2024.2.105 (pre-release)
-   OS and version: macOS 14.4 (23E214)
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.11.4.

## Problem

I think Pylance should only use pyright settings in `pyproject.toml` (specifically `include`) for pyright related functionality, not for functionality unrelated to pyright such as refactoring. In my case specifically, I want to use pyright only on a subset of my code while I want refactoring functionality such as _Move Symbol_ and _Rename Symbol_ to be available in the whole code base (and not failing silently on the part that's not in `include`).

## Expected behavior

A pyright setting in `pyproject.toml` does not affect pylance behaviour unrelated to pyright, i.e. renaming symbols is unaffected by a pyright setting.

## Actual behaviour

Setting the `include` key for pyright in `pyproject.toml` limits pylance _Rename Symbol_ and _Move Symbol_ functionality to that very directory too. 

### Reprex
I created a [demo](https://github.com/lorenzwalthert/vs-code-refactoring-bug) repo to show the behaviour: The repo contains a python source package in `vs_code_refactoring_bug/`:

* `vs_code_refactoring_bug/core.py` defines a function called `map`. 
* `vs_code_refactoring_bug/using_core.py` imports `map`.

At the [initial commit](https://github.com/lorenzwalthert/vs-code-refactoring-bug/tree/c132c3cce9755db1697e6795609eca8019da8cf4) of the repo, _Rename Symbol_ call works as expected, i.e. renaming the object happens in all other files as well. Here a screenshot from renaming with new refactor preview
<img width="424" alt="Screenshot 2024-03-08 at 21 37 51" src="https://github.com/microsoft/pylance-release/assets/10477073/a28854b5-63bc-4b64-883c-3b8d237afa69">

And here are the files affected with refactor preview:

<img width="508" alt="Screenshot 2024-03-08 at 21 38 18" src="https://github.com/microsoft/pylance-release/assets/10477073/8c7dd493-f47b-47b1-86f7-9b18ea4f350c">

The [second commit](https://github.com/lorenzwalthert/vs-code-refactoring-bug/commit/8a28a62094674d2920d8109c92c67520e33c02cd) adds the following to the `pyproject.toml`:
```toml
[tool.pyright]
include = ["vs_code_refactoring_bug/src/"]
```
And the _Rename Symol_ now only renames in the file that was selected, not other files. The lightbulb next to the declaration does not appear anymore (Screenshots omitted here for brievity).

**Comment by lorenzwalthert** [comment]
To be fair, setting `python.analysis.include to `[]` also solves the problem. Apparently that's the default, but who knew setting it to the default would help.

**Comment by lorenzwalthert** [comment]
Setting [`python.analysis.include`](https://marketplace.visualstudio.com/items?itemName=ms-python.vscode-pylance) to `[]` or `["**"]` does not solve the problem either.

**Comment by debonte** [comment]
This is by design. Pylance's refactoring features are heavily dependent on Pyright's analysis (ex. code flow analysis).

**Comment by lorenzwalthert** [comment]
Ok, I guess I could have spared the reprex 😀.

Point is that I want to run type checking (with pre-commit hook) only on a subset of my packages, and refactoring tools should take effect on all python files in the repo. While I see that pyright and pylance things work arm in arm, for the user experience, that is not very desirable in my case. Also, I don't want static analysis to be super slow. 

Hence, I used `include`, `exclude` though `pyproject.toml` for my editor experience and for pre-commit hook, I changed my `entrypoint` to 
```python
poetry run pyright path/to/sub/package/**/*.py
```
which overrides the settings from `pyproject.toml`.
